tests/test_machine.py

```diff
@@ -1,69 +1,82 @@
 # SPDX-License-Identifier: LGPL-2.1+
 
-import os
-from subprocess import TimeoutExpired
-
+import unittest
 import pytest
+from subprocess import TimeoutExpired
+from sys import exc_info
+from typing import List, Tuple, Optional, Type
+from types import TracebackType
 
-import mkosi.machine as machine
 from mkosi.backend import MkosiException
-
-
-pytestmark = [
-    pytest.mark.integration,
-    pytest.mark.parametrize("verb", ["boot", "qemu"]),
-    pytest.mark.skipif(os.getuid() != 0, reason="Must be invoked as root.")
-]
-
-
-def test_simple_run(verb: str) -> None:
-    with machine.Machine([verb]) as m:
-        p = m.run(["echo", "This is a test."])
-        assert "This is a test." == p.stdout.strip("\n")
-
-    assert m.exit_code == 0
-
-
-def test_wrong_command(verb: str) -> None:
-    # First tests with argument check = True from mkosi.backend.run(), therefore we see if an exception is raised
-    with machine.Machine([verb]) as m:
+from mkosi.machine import Machine
+
+
+class MkosiMachineTest(unittest.TestCase):
+    reboot: bool
+    args: List[str]
+    machine: Machine
+    exc_info: Tuple[Optional[Type[BaseException]], Optional[BaseException], Optional[TracebackType]]
+
+    def _callTestMethod(self, method: str) -> None:
+        try:
+            super()._callTestMethod(method)
+        except Exception as e:
+            self.exc_info = exc_info()
+            raise e
+
+    def __init_subclass__(cls, args: List[str], reboot: bool = True) -> None:
```

Comment:
Like we discussed, let's not make `reboot` configurable. We always want to reboot so that tests cannot accidentally depend on the result of previous tests which makes adding concurrency in the future much harder.

Reply:
Also, `args` can be `Sequence` instead of `List` I think